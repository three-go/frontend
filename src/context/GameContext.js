import React, { createContext, useState } from "react";

const GameContext = createContext();

export const GameContextProvider = ({ children }) => {
  const [currentGameKey, setCurrentGameKey] = useState("");

  return (
    <GameContext.Provider value={{ currentGameKey, setCurrentGameKey }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContext;
