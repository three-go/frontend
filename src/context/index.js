import GameContext, { GameContextProvider } from "./GameContext";

export { GameContext, GameContextProvider };
