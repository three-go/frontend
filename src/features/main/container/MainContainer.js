import React, { useState, useEffect, useContext } from "react";

import { Platform } from "react-native";
import RNExitApp from "react-native-exit-app";
import { SafeAreaProvider } from "react-native-safe-area-context";
import SystemNavigationBar from "react-native-system-navigation-bar";

import { DefaultButton, SmallButton, ContentModal } from "../../../components";
import { GameContext } from "../../../context";
import { game } from "../../../utils";
import Main from "../presenter/Main";

const MainContainer = ({ navigation }) => {
  const { currentGameKey, setCurrentGameKey } = useContext(GameContext);

  const [scoreModalVisible, setScoreModalVisible] = useState(false);
  const [descriptionModalVisible, setDescriptionModalVisible] = useState(false);

  useEffect(() => {
    const hideSoftKey = async () => {
      await SystemNavigationBar.navigationHide();
    };

    if (Platform.OS === "android") {
      hideSoftKey();
    }
  }, []);

  const exitApp = () => {
    RNExitApp.exitApp();
  };

  const handleShowScoreModal = () => {
    setScoreModalVisible(!scoreModalVisible);
  };

  const handleShowDescriptionModal = () => {
    setDescriptionModalVisible(!descriptionModalVisible);
  };

  const handleSetGameNameAndShowDescriptionModal = (name) => {
    return () => {
      setCurrentGameKey(name);
      setDescriptionModalVisible(!descriptionModalVisible);
    };
  };

  const handleStartGame = () => {
    setDescriptionModalVisible(!descriptionModalVisible);
    navigation.navigate("Game1");
  };

  console.log("----------------------------------");
  console.log("currentGameKey", currentGameKey);
  console.log("----------------------------------");

  return (
    <SafeAreaProvider>
      <Main
        onExitApp={exitApp}
        handleShowScoreModal={handleShowScoreModal}
        handleSetGameNameAndShowDescriptionModal={
          handleSetGameNameAndShowDescriptionModal
        }
      />

      {scoreModalVisible && (
        <ContentModal
          title="점수 현황"
          content="김춘식 : 500점"
          isVisible={scoreModalVisible}
        >
          <DefaultButton
            content="닫기"
            color="#c92a2a"
            onPress={handleShowScoreModal}
          />
        </ContentModal>
      )}

      {descriptionModalVisible && (
        <ContentModal
          title="게임 설명"
          content={game.description[currentGameKey]}
          isVisible={descriptionModalVisible}
        >
          <SmallButton
            content="게임 시작"
            color="#00BBD1"
            onPress={handleStartGame}
          />
          <SmallButton
            content="닫기"
            color="#c92a2a"
            onPress={handleShowDescriptionModal}
          />
        </ContentModal>
      )}
    </SafeAreaProvider>
  );
};

export default MainContainer;
