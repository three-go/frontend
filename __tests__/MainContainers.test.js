import React from "react";

import { render } from "@testing-library/react-native";

import "@testing-library/jest-native";
import "react-native-gesture-handler/jestSetup";
import { GameContextProvider } from "../src/context/GameContext";
import MainContainer from "../src/features/main/container/MainContainer";

const mockNavigate = jest.fn();
const mockDisPatch = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockNavigate,
      dispatch: mockDisPatch,
    }),
    useRoute: () => ({}),
  };
});

jest.mock("react-native-reanimated", () => {
  const Reanimated = require("react-native-reanimated/mock");
  Reanimated.default.call = () => {};
  return Reanimated;
});

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

jest.mock("react-native/Libraries/EventEmitter/NativeEventEmitter.js", () =>
  require("react-native/Libraries/EventEmitter/__mocks__/NativeEventEmitter.js")
);

jest.mock("react-native-sound", () => {
  const Sound = () => {
    const obj = {};
    obj.sound = "music";
    obj.setNumberOfLoops = jest.fn();
    obj.play = jest.fn();
    return obj;
  };

  return Sound;
});

jest.mock("@react-native-community/async-storage", (key, value) => {
  const storage = {};
  storage[key] = value;
  return storage;
});

describe("Container Test", () => {
  it("MainContainer Test", () => {
    const mainContainer = render(
      <GameContextProvider>
        <MainContainer />
      </GameContextProvider>
    );
    const mainContainerJson = mainContainer.toJSON();

    expect(mainContainer.getByTestId("mainContainer")).toBeTruthy();
    expect(mainContainerJson).toBeTruthy();
    expect(mainContainerJson).toMatchSnapshot();
  });
});
