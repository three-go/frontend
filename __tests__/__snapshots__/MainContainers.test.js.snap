// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Test MainContainer Test 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="mainContainer"
/>
`;
